tests/test_warcprox.py

```diff
@@ -2106,6 +2106,40 @@ def test_crawl_log(warcprox_, http_daemon, archiving_proxies):
     extra_info = json.loads(fields[12].decode('utf-8'))
     assert set(extra_info.keys()) == {'exception'}
 
+    #Verify non-ascii urls are encoded properly
```

Comment:
Thanks for expanding test coverage!

A couple general comments (I'm not familiar enough with the behavior under test to leave more specific feedback):

- would it be possible to break this test case out into its own test function? 
- Consider being more targeted in the assertions you make here. It's not clear to me what key behavior would break if non-ascii urls weren't encoded properly. If all the assertions are in fact important, then I'd encourage adding a bit of commentary about what's happening and how it's related to the behavior under test. There's the "Given/When/Then" strategy which I feel helps elucidate the behaviors targeted by specific test cases: https://martinfowler.com/bliki/GivenWhenThen.html ; this could be a nice thing to try. Here's an IA-specific example: https://git.archive.org/dps/vault-site/blob/13f9c3cdf4881d2e409d6c5f8ff068e42ec27d85/tests/vault/models_tests.py#L253